tests/run-make/thumb-none-qemu/rmake.rs

```diff
@@ -0,0 +1,51 @@
+//! How to run this
+//! $ ./x.py clean
+//! $ ./x.py test --target thumbv6m-none-eabi,thumbv7m-none-eabi tests/run-make
```

Comment:
Suggestion:

- It would be really helpful if we explained what this test is trying to test... From what I can tell, this is just a basic smoke test for `thumbv6m-none-eabi` and `thumbv7m-none-eabi` to see if cargo + rustc can cross-compile for them?
- Also I have no clue if these test instructions are accurate. In the example crate, there's a

```rs
// exit from qemu
semihosting::debug::exit(semihosting::debug::EXIT_SUCCESS);
```

I have no clue if this actually can only be run via qemu...

Reply:
well normally embedded programs don't really exit: their main function is just an infinite loop. Just for testing purposes we need to indicate that main is done, which is done with that line. But that won't work for production programs on physical hardware.